product.php

```diff
@@ -5,23 +5,34 @@
 use Furniture\Hydrators\ProductHydrator;
 use Furniture\Services\ResponseService;
 use \Furniture\Exceptions\InvalidProductException;
+use \Furniture\Entities\DetailedProductEntity;
+use \Furniture\Exceptions\InvalidUnitException;
+use \Furniture\Services\MeasurementConverterService;
 
 const SUCCESS_MESSAGE = "Successfully retrieved product";
 
 $productId = $_GET['id'] ?? '';
+$unit = $_GET['unit'] ?? 'mm';
+$validUnits = MeasurementConverterService::VALID_UNITS;
 
 try {
     if (!is_numeric($productId)) {
         throw new InvalidProductException(InvalidProductException::INVALID_PROD_ID);
     }
+    if (!in_array($unit, $validUnits)) {
```

Comment:
I think it would be a better separation of concerns if this validation happens in the DetailedProductEntity class when you try to set its measurement unit.